common/expression/src/types/string.rs

```diff
@@ -227,4 +227,71 @@ impl StringColumnBuilder {
         assert_eq!(self.offsets.len(), 2);
         self.data[(self.offsets[0] as usize)..(self.offsets[1] as usize)].to_vec()
     }
+
+    pub fn from_scalar(data: Vec<u8>) -> Self {
+        StringColumnBuilder {
+            offsets: vec![0, data.len() as u64],
+            data,
+        }
+    }
+
+    pub fn try_from_transform_scalar<F>(
+        val: &[u8],
+        estimate_bytes: usize,
+        mut func: F,
+    ) -> Result<StringColumnBuilder, String>
+    where
+        F: FnMut(&[u8], &mut [u8]) -> Result<usize, String>,
+    {
+        let mut buf = Vec::with_capacity(estimate_bytes);
+        unsafe {
+            let bytes = std::slice::from_raw_parts_mut(buf.as_mut_ptr(), buf.capacity());
+            let len = func(val, bytes)?;
+            buf.set_len(len);
+        }
+
+        Ok(StringColumnBuilder {
+            offsets: vec![0, buf.len() as u64],
```

Comment:
So we do not need this offsets vec for scalar.

Just return `Ok(buf)`

Reply:
You are right, the offsets is no need here.
let me remove this.